setup/src/opts.ts

```diff
@@ -93,6 +93,7 @@ export function getOpts(
   const stackSetupGhc = (inputs['stack-setup-ghc'] || '') !== '';
   const stackEnable = (inputs['enable-stack'] || '') !== '';
   const matcherDisable = (inputs['disable-matcher'] || '') !== '';
+  const cabalUpdate = inputs['cabal-update'] !== 'false';
```

Comment:
Not sure if this is the customary check for setting a default `true` input to `false`.

Reply:
hmm i would follow the handling of other boolean params, for consistency shake

Reply:
from https://github.com/haskell/actions/blob/main/setup/README.md

> Note: "boolean" types are set/unset, not true/false. That is, setting any "boolean" to a value other than the empty string ("") will be considered true/set.

dont like it at all, cause seeting it to `false` make the option `true` but 🤷

Reply:
So maybe it needs a `disable-cabal-update`? like `disable-matcher`? omg 🤦

Reply:
@jneira: Thanks for digging into this!  
I also dislike the current policy.  Like you said, `disable-matcher: false` currently seems to _disable_ the matcher, and `enable-stack: false` would _enable_ stack.  This will definitely confuse readers of workflows, and opens the door to obfuscation.

Is `true`/`false` the only accepted way to set booleans in YAML/JSON?  What about `True`/`False`, `on`/`off`, `yes`/`no`?  Maybe the authors of this action wanted to avoid such minefields...

I wonder whether there isn't some core functionality for actions to parse boolean inputs in a robust way, so that we do not have to reinvent the wheel here.

Reply:
There seems to be [`JSON.parse()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse) which accepts `true` and `false` and throws `SyntaxError`.

Reply:
yeah, another alternative could be extends the Default data type to track explicitly option default values and set to true cabal-uodate

 I would left changing options semantic to another pr, cause it will break compatibility

Reply:
Oh, there is a [`core.getBooleanInput(...)`](https://github.com/actions/toolkit/blob/b2d865f18051b214475dae41766e9970fd68ca12/packages/core/src/core.ts#L163-L183) [added in 2021](https://github.com/actions/toolkit/pull/725).

Reply:
I settled for using a modified version of `getBooleanInput`: `parseYAMLBoolean`.